samples/net/sockets/coap_client/src/coap-client.c

```diff
@@ -42,7 +42,7 @@ static struct coap_block_context blk_ctx;
 
 static void wait(void)
 {
-	if (poll(fds, nfds, K_FOREVER) < 0) {
+	if (poll(fds, nfds, K_MS_FOREVER) < 0) {
```

Comment:
I used NET_WAIT_FOREVER symbol in the latest version of this PR but that can be changed if we decide that plain -1 is to be used.

Reply:
With `poll` being a standard POSIX function and the standard clearly stating that the timeout value of `-1` means "forever," shouldn't it be perfectly okay to use `-1` here?

> If the value of timeout is -1, poll() shall block until a requested event occurs or until the call is interrupted.

https://pubs.opengroup.org/onlinepubs/9699919799/functions/poll.html

Reply:
Agreed. The value of NET_WAIT_FOREVER is also -1 so that could be used too.

Reply:
> I used NET_WAIT_FOREVER symbol in the latest version of this PR but that can be changed if we decide that plain -1 is to be used.

Looking at https://github.com/zephyrproject-rtos/zephyr/pull/24149/files#diff-e986daadfca28716ec435fc4c722ae40L28 , that indeed appears to be the plan. Using different ways to specify the same behavior for different APIs in Zephyr would be rather confusing. Maybe @pabigot as the author of #24149 can comment here.

Otherwise, as @stephanosio points out, as long as values used are consistent with "well-known APIs" (so, the values themselves are well-known), I think it's fine to skip symbolic constants and use these well-known, intuitive (enough) values directly. After all, we don't write `TWO + TWO`.

@jukkar, Also compare our discussion on a similar matter https://github.com/zephyrproject-rtos/zephyr/pull/24124#pullrequestreview-388278757 . Of all other matters, consistency is important. That's why I used "-1" in a sample config to mean "forever"/"infinite", and let 0 mean 0 (skipping not pertinent discussion whether 0 makes sense or not in a particular case or nor - if it doesn't, users just won't use it there). And surely, we can have different representations in different places - if there're good reasons to do so. If not, consistency should prevail.

Reply:
@pfalcon Regarding #24149, originally the thread start delay parameter in `K_THREAD_DEFINE` was specified in milliseconds.  At the time we started moving from integral millisecond timeouts to `k_timeout_t` (last fall), the API changed to take a `k_timeout_t` value for consistency with `k_thread_create()`, so 0 was converted to `K_NO_WAIT` and -1 to `K_FOREVER`.  When the delay changed back to milliseconds (this spring) forever was converted back to -1, which may be just a canonical representation for an unlimited wait specified by a negative value (I haven't looked).

The `K_THREAD_DEFINE` API does not actually document support for "wait-forever" (never start it automatically) regardless of that being useful, and continues to recommend `K_NO_WAIT` for no-delay even though that is now wrong as it's a timeout value.